### PR TITLE
Malky notification

### PR DESCRIPTION
**Pull request by MalkySinowitz**
display notification from extension

### CODE REVIEW

File: extension-ui/background.js [hunk 1]
```diff
@@ -82,3 +82,28 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
     return true; // כדי להורות שהתגובה היא אסינכרונית
   }
```
Comment (mindas2000) [inline extension-ui/background.js:83]: please remove the comment

File: server-side/email-service/app/consumer.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+consumer.on('error', (err) => {
+  console.error('Consumer error:', err);
+});
+
+const handleEvent = (event) => {
+  switch (event.type) {
+    case 'sendEmail':
+      sendEmail(event.payload);
+      break;
+    default:
```
Comment (mindas2000) [inline server-side/email-service/app/consumer.js:25]: If/else is not enough for you, you just made only one option

File: server-side/email-service/app/mailer.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import dotenv from 'dotenv';
+dotenv.config();
+import nodemailer from 'nodemailer';
+import Bottleneck from 'bottleneck';
+import retry from 'async-retry';
+
+let transporter = nodemailer.createTransport({
+  service: 'gmail',
```
Comment (mindas2000) [inline server-side/email-service/app/mailer.js:8]: why is not it const?

File: server-side/email-service/app/producer.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+admin.createTopics(topicToCreate, async (err, res) => {
+  if (err) {
+    console.error('Failed to create topics:', err);
+  } else {
+  }
+});
+
+const producer = new Producer(client);
+const email='malkysino@gmail.com'
```
Comment (mindas2000) [inline server-side/email-service/app/producer.js:26]: please add ; in this line and in the following lines too

File: server-side/email-service/app/producer.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+const subject='אלופות'
+const text='אם זה הגיע הפרויקט עובד🤣😘😂😊😍😁אין עלינו!!!!!!!!!!'
```
Comment (mindas2000) [inline server-side/email-service/app/producer.js:28]: apparently you will have to remove it

File: extension-ui/background.js [hunk 6]
```diff
@@ -82,3 +82,28 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
...
+      type: 'basic',
+      iconUrl: 'images/icon48.png',
+      title: title,
+      message: message,
+      priority: 2
+  }, function(notificationId) {
+      if (chrome.runtime.lastError) {
+          console.error(chrome.runtime.lastError);
+      } else {
+          console.log('Notification created with ID:', notificationId);
```
Comment (mindas2000) [inline extension-ui/background.js:97]: console.log it's importent?

File: server-side/email-service/app/consumer.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import dotenv from 'dotenv';
+dotenv.config();
```
Comment (mindas2000) [inline server-side/email-service/app/consumer.js:2]: move it after the import, it seems standards to me

File: server-side/email-service/app/mailer.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import dotenv from 'dotenv';
+dotenv.config();
```
Comment (mindas2000) [inline server-side/email-service/app/mailer.js:2]: move it after the import, it seems standards to me

File: server-side/email-service/app/producer.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
+import dotenv from 'dotenv';
+dotenv.config();
```
Comment (mindas2000) [inline server-side/email-service/app/producer.js:2]: move it after the import, it seems standards to me

File: server-side/email-service/app/run.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import dotenv from 'dotenv';
+dotenv.config();
```
Comment (mindas2000) [inline server-side/email-service/app/run.js:2]: move it after the import, it seems standards to me

File: extension-ui/background.js (hunk 1 above)
Comment (MalkySinowitz) [reply extension-ui/background.js:83]: Not relevant to my code

**Review by mindas2000 — APPROVED** [review]
I did not find  console log, lines ending without ; , variables with unstructured names etc

File: extension-ui/background.js [hunk 11]
```diff
@@ -82,3 +82,23 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
     return true; // כדי להורות שהתגובה היא אסינכרונית
   }
 });
+
+function showNotification(title, message) {
+  chrome.notifications.create({
+      type: 'basic',
```
Comment (SariGlick) [inline extension-ui/background.js:88]: please change to notification that contains the text:
Notice: 
browsing time in {site} is about to reach limitation in {num} minutes.

the texts should be saved in constants file. site & num are parameters.

File: extension-ui/background.js [hunk 12]
```diff
@@ -82,3 +82,23 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
...
+      iconUrl: 'images/icon48.png',
+      title: title,
+      message: message,
+      priority: 2
+   }
+);
+}
+
+chrome.tabs.onCreated.addListener((tab) => {
```
Comment (SariGlick) [inline extension-ui/background.js:97]: please remove the examples of the notifications in opening and closing

File: extension-ui/manifest.json [hunk 13]
```diff
@@ -20,6 +21,11 @@
   "action": {
     "default_popup": "popup.html"
   },
+  "icons": {
+    "16": "images/icon16.png",
```
Comment (SariGlick) [inline extension-ui/manifest.json:25]: please remove this, and the icon files as well

File: server-side/email-service/Docker-compose.yml [hunk 14]
```diff
@@ -0,0 +1,32 @@
+version: '3.7'
+services:
```
Comment (SariGlick) [inline server-side/email-service/Docker-compose.yml:2]: please remove all files related to emails service

**Review by SariGlick — CHANGES_REQUESTED** [review]

**Pull request closed without merging**